tfe/data_source_oauth_clients_test.go

```diff
@@ -0,0 +1,68 @@
+package tfe
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccTFEOAuthClientsDataSource_basic(t *testing.T) {
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			if GITHUB_TOKEN == "" {
```

Comment:
What do you think about moving checking the existence of `GITHUB_TOKEN` to the top of the method body to trigger an early exit and avoid a few method calls?